src/main/java/org/jenkinsci/test/acceptance/docker/DockerContainer.java

```diff
@@ -118,6 +118,25 @@ public int port(int n) {
         }
     }
 
+    /**
+     * Finds the ephemeral UDP port that the given container port is mapped to.
+     */
+    public int udpPort(int n) {
+        assertRunning();
+        try {
+            if (sharingHostDockerService()) {
+                return n;
+            }
+            String out = Docker.cmd("port").add(cid, n + "/udp").popen().verifyOrDieWith("docker port command failed").trim();
+            if (out.isEmpty())  // expected to return single line like "0.0.0.0:55326"
+                throw new IllegalStateException(format("Udp port %d is not mapped for container %s", n, cid));
+
+            return Integer.parseInt(out.split(":")[1]);
```

Comment:
And then you are throwing away the first component. Surely you need a UDP equivalent of `ipBound`?

Reply:
@jglick I added the UDP equivalent for `ipBound`, but I don't need it. The reason why I don't need this is that the samba container is reachable inside the [build-docker-container](https://github.com/fbelzunc/active-directory-plugin-build-container-with-docker-fixtures) from the IP address `127.0.0.1` - all containers which share the same Docker Engine shared this behavior and it is likely not to change.